_RELEASE/Packs/cube/Scripts/utils.lua

```diff
@@ -1,21 +1,552 @@
+end
+
+function easeOutSine(x)
+	x = clamp(x, 0, 1);
+	return math.sin((x * math.pi) / 2);
+end
+
+-- Quadratic
+function easeInQuad(x)
+	x = clamp(x, 0, 1);
+	return x ^ 2;
+end
+
+function easeInOutQuad(x) 
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return 2 * x * x;
+	end
+	return 1 - (-2 * x + 2) ^ 2 / 2;
+end
+
+function easeOutQuad(x)
+	x = clamp(x, 0, 1);
+	return 1 - (1 - x) * (1 - x);
+end
+
+-- Cubic
+function easeInCubic(x)
+	x = clamp(x, 0, 1);
+	return x ^ 3;
+end
+
+function easeInOutCubic(x)
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return 4 * x * x * x;
+	end
+	return 1 - (-2 * x + 2) ^ 3 / 2;
+end
+
+function easeOutCubic(x)
+	x = clamp(x, 0, 1);
+	return 1 - (1 - x) ^ 3;
+end
+
+-- Quartic
+function easeInQuart(x)
+	x = clamp(x, 0, 1);
+	return x ^ 4;
+end
+
+function easeInOutQuart(x)
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return 8 * x * x * x * x;
+	end
+	return 1 - (-2 * x + 2) ^ 4 / 2; 
+end
+
+function easeOutQuart(x)
+	x = clamp(x, 0, 1);
+	return 1 - (1 - x) ^ 4;
+end
+
+-- Quintic
+function easeInQuint(x)
+	x = clamp(x, 0, 1);
+	return x ^ 4;
+end
+
+function easeInOutQuint(x)
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return 16 * x * x * x * x * x;
+	end
+	return 1 - (-2 * x + 2) ^ 5 / 2;
+end
+
+function easeOutQuint(x)
+	x = clamp(x, 0, 1);
+	return 1 - (1 - x) ^ 5;
+end
+
+-- Exponential
+function easeInExpo(x)
+	x = clamp(x, 0, 1);
+	if (x == 0) then
+		return 0;
+	end
+	return 2 ^ (10 * x - 10);
+end
+
+function easeInOutExpo(x)
+	x = clamp(x, 0, 1);
+	if (x == 0) then
+		return 0;
+	elseif (x < 0.5) then
+		return 2 ^ (20 * x - 10) / 2;
+	elseif (x == 1) then
+		return 1;
+	end
+	return (2 - 2 ^ (-20 * x + 10)) / 2;
+end
+
+-- Easings.net has this formula WRONG. This is the actual formula right here.
+function easeOutExpo(x)
+	x = clamp(x, 0, 1);
+	if (x == 1) then
+		return 1;
+	end
+	return -2 ^ (-10 * x) + 1;
+end
+
+-- Circle
+function easeInCirc(x)
+	x = clamp(x, 0, 1);
+	return 1 - math.sqrt(1 - (x^2));
+end
+
+function easeInOutCirc(x)
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return (1 - math.sqrt(1 - (2 * x) ^ 2)) / 2;
+	end
+	return (math.sqrt(1 - (-2 * x + 2) ^ 2) + 1) / 2;
+end
+
+function easeOutCirc(x)
+	x = clamp(x, 0, 1);
+	return math.sqrt(1 - (x - 1) ^ 2);
+end
+
+-- Back Functions
+function easeInBack(x)
+	x = clamp(x, 0, 1);
+	local CONST_ONE = 1.70158;
+	local CONST_THREE = CONST_ONE + 1;
+
+	return CONST_THREE * x * x * x - CONST_ONE * x * x
+end
+
+function easeInOutBack(x)
+	x = clamp(x, 0, 1);
+	local CONST_ONE = 1.70158;
+	local CONST_TWO = CONST_ONE * 1.525;
+
+	if (x < 0.5) then
+		return ((2 * x) ^ 2 * ((CONST_TWO + 1) * 2 * x - CONST_TWO)) / 2;
+	end
+	return ((2 * x - 2) ^ 2 * ((CONST_TWO + 1) * (x * 2 - 2) + CONST_TWO) + 2) / 2;
+end
+
+function easeOutBack(x)
+	x = clamp(x, 0, 1);
+	local CONST_ONE = 1.70158;
+	local CONST_THREE = CONST_ONE + 1;
+
+	return 1 + CONST_THREE * (x - 1) ^ 3 + 
+			CONST_ONE * (x - 1) ^ 2;
+end
+
+-- Elastic
```

Comment:
Remove semicolons, they are redundant in general-purpose but for one line writing like several functions/variables in one line, etc. It's not java or C++ and it doesn't follow the style of the language.

Reply:
You're honestly nitpicking with this. Semicolons have zero effect on the code itself, and I do it because I can. It's not like it negatively affects the performance of the code.

Reply:
Well then I will not follow your clang-formatting nitpicks if you don't want to follow Lua's style of coding lmao.

Reply:
It's so very really true that semicolons really blend well with your python-style comments 🤔

Reply:
Now you two are just being flat out childish. This is like the "indentation" argument we had in the Discord server. The difference being that indentation actually matters because it affects the legibility of the code and can actually make a huge difference in how it is done. The semicolon is just a single little character at the end of a line and shouldn't affect legibility at all (unless your compressing multiple lines into one). You can easily pretend that it isn't there. I just do this so that way I don't get a bad habit of forgetting semicolons when I go to a language where semicolons are actually required to end lines (Ahem, C++). Again, Lua has it so you can put semicolons at the end of a line if you want it to; that's the choice I am deciding to make; get over it.

Reply:
Well idk i personally don't forget how to use languages I've learned and i try to follow their standard and not mix it up cuz haha habbits.

Reply:
but I don't really care that much lol. But it still looks kind of stupid.

Reply:
I mean to be fair it's not the first time both of you have been fighting over nitpicks, I couldn't help joining in to throw my stupid sarcastic two cents but at the end of the day this is just up to preference so whatever.

Reply:
> Well then I will not follow your clang-formatting nitpicks if you don't want to follow Lua's style of coding lmao.

I am not doing this because I personally enjoy the format. That's how Vee wants the code to be formatted. Vee would do the exact same thing where he'll ask you to fix the formatting issues. Again, why do you have to act like a child?

Reply:
Because I guess I don't like you this much I don't notice it lol.
But my point stands about inconsitency.